src/wordlist.h

```diff
@@ -65,5 +67,7 @@ char *wordlistChopHead(wordlist **);
 /// convert a wordlist to a SBufList
 SBufList ToSBufList(wordlist *);
 
+std::ostream& operator<<(std::ostream &os, const wordlist wl);
```

Comment:
If possible, please specialize AsList<wordlist> instead. AFAICT, doing so will fix at least three problems with the current PR code, offer more formatting flexibility to callers, and reduce future code changes related to wordlist elimination from Squid.

When specializing, please try to avoid duplicating the existing manipulator.container [loop](https://github.com/squid-cache/squid/blob/6410d312077b71a1f5f619390c2d467b1fe4dc74/src/base/IoManip.h#L203-L213). Instead, try adding worldlist::begin(), worldlist::end() (at least).

Reply:
Done.

Build-tested with this (why, oh why doesn't github support stacked diffs!)

```
diff --git a/src/errorpage.cc b/src/errorpage.cc
index a56998dd72..5f35a5e44a 100644
--- a/src/errorpage.cc
+++ b/src/errorpage.cc
@@ -10,6 +10,8 @@
 
 #include "squid.h"
 #include "AccessLogEntry.h"
+#include "base/IoManip.h"
+#include "base/PackableStream.h"
 #include "cache_cf.h"
 #include "clients/forward.h"
 #include "comm/Connection.h"
@@ -864,7 +866,9 @@ ErrorState::Dump(MemBuf * mb)
         str.appendf("FTP Request: %s\r\n", ftp.request);
         str.appendf("FTP Reply: %s\r\n", (ftp.reply? ftp.reply:"[none]"));
         str.append("FTP Msg: ", 9);
-        wordlistCat(ftp.server_msg, &str);
+        PackableStream s(str);
+        s << AsList(*(ftp.server_msg));
+        // wordlistCat(ftp.server_msg, &str);
         str.append("\r\n", 2);
     }
 ```